Sources/Datadog/DatadogConfiguration.swift

```diff
@@ -74,19 +74,31 @@ extension Datadog {
             }
         }
 
-        internal let clientToken: String
-        internal let environment: String
-        internal let loggingEnabled: Bool
-        internal let tracingEnabled: Bool
-        internal let rumEnabled: Bool
-        internal let logsEndpoint: LogsEndpoint
-        internal let tracesEndpoint: TracesEndpoint
-        internal let rumEndpoint: RUMEndpoint
-        internal let serviceName: String?
-        internal let tracedHosts: Set<String>
-        internal let rumSessionsSamplingRate: Float
+        /// The RUM Application ID.
+        private(set) var rumApplicationID: String?
+        /// Either the RUM client token (which supports RUM, Logging and APM) or regular client token, only for Logging and APM.
+        private(set) var clientToken: String
+        private(set) var environment: String
+        private(set) var loggingEnabled: Bool
+        private(set) var tracingEnabled: Bool
+        private(set) var rumEnabled: Bool
+        private(set) var logsEndpoint: LogsEndpoint
+        private(set) var tracesEndpoint: TracesEndpoint
+        private(set) var rumEndpoint: RUMEndpoint
+        private(set) var serviceName: String?
+        private(set) var tracedHosts: Set<String>
+        private(set) var rumSessionsSamplingRate: Float
 
-        /// Creates configuration builder and sets client token.
+        /// Creates the builder for configuring the SDK to work with RUM, Logging and Tracing features.
+        /// - Parameter rumApplicationID: RUM Application ID obtained on Datadog website.
+        /// - Parameter rumClientToken: RUM Client Token (generated for the RUM Application ID) obtained on Datadog website.
+        /// - Parameter environment: the environment name which will be sent to Datadog. This can be used
+        ///  to filter events on different environments (e.g. "staging" or "production").
+        public static func builderUsing(rumApplicationID: String, rumClientToken: String, environment: String) -> Builder {
```

Comment:
oh so you do provide a static builder factory method already...maybe you should think about what I mentioned below.